### DISCUSSION TITLE
Forum to ask questions

### DISCUSSION THREAD

**Discussion by jrheisey** [rocky/remake#140] [General]
Where can I post questions regarding remake?

**Comment by rocky** [comment]
I suppose here in discussions is fine. _Edit: provided it is about remake rather than how do I debug my Makefile setup (whether or  not you use remake in the process)_

**Reply by jrheisey** [reply]
I am looking for guidance on how to debug a specific situation in a complicated set of make files.

Specifically when an intermediate target is not being built after I delete only it.

### Overview of a typical scenerio
- A subfolder is using protoc to build a dot cc file from a dot proto file
- That is turn is compiled to an object file along with other dot cc files
- The resulting target is a static library
### Issue
- When all targets are cleaned all targets are built as expected
- If I then delete the generated dot cc and or any of the object files they won't rebuild
### My efforts
- I am analyzing the database and trace logs
- I created a simplified version of the make file which works as expected
- I'm comparing the working set with the failing set of database and trace logs
- I tried setting breaks but I don't know how to determine why a file was not built
### Requested Guidance
- Can you reference material that helps me understand the database and trace logs?
- Can you explain to me what I should and should not be seeing in the database and trace logs?

I hope that description is clear enough

**Reply by rocky** [reply]
This  doesn't seem to to be about `remake` specifically but rather a request for help in "how to debug a specific situation in a complicated set of make files."

Stackoverflow or something like that is a more appropriate place for this kind of general solicitation of help.

**Reply by jrheisey** [reply]
I was expectng that remake provided relevant information or methodology which standard make does not for these types of issues

**Reply by jrheisey** [reply]
HAH I didn't expect such a quick response. :)

**Reply by jrheisey** [reply]
Is remake then more suited to debug the recipes?

**Reply by jrheisey** [reply]
Is there a forum specifically for make?

**Reply by rocky** [reply]
> I was expectng that remake provided relevant information 

Possibly, but your question isn't about `remake` except in a very superficial way.  I have no doubt that there are those who have ideas about the issues and topics and do not use `remake` but GNU make instead. 

Here is an analogy.  Suppose I write a debugger that I think has features that other debuggers don't have. (Well, actually, I have.) 

This doesn't mean that any kind of specific debugging problem or philosophical discussion of debugging in that programming language is a topic for that debugger.  There needs to be some specific feature of  `remake` that you tried for it to be something of discussion here. 

On the other hand if you've been currently using remake to do something and there is a particular feature about remake that is not part of GNU Make and you have an observation or want to discuss some aspect of that feature, well, then a discussion here is fine.

**Reply by rocky** [reply]
> Is there a forum specifically for make?

Might be an old newsgroup. Look at https://groups.google.com

**Reply by rocky** [reply]
> Is remake then more suited to debug the recipes?

You seem to ask a lot of questions. This just is not the right place for these kinds of questions.

I hope you find the right place and get the guidance you are seeking.
